idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/lineMarker/LineMarkerSyntax.kt

```diff
@@ -244,6 +246,25 @@ interface LineMarkerSyntax {
       }
     )
 
+  fun IdeMetaPlugin.addLineMarkerProviderM(
+    icon: Icon,
+    transform: (PsiElement) -> KtValueArgument?,
```

Comment:
There is a way to abstract this to any `A : PsiElement` and add a another type variable `B` for the specified child. 
`e.g.: child : Class<B>` instead of `LeafPsiElement::class.java`.
Then we can reuse this for other use-cases.

Reply:
I just noticed I can remove it completely.. and re-use the one above by tweaking it a little

Reply:
That's even better :)